src/pages/ReimbursementAccount/ACHContractStep.js

```diff
@@ -36,74 +38,92 @@ const propTypes = {
         if (this.state.hasOtherBeneficialOwners) {
-            beneficialOwnersErrors = _.map(this.state.beneficialOwners, ValidationUtils.validateIdentity);
+            _.each(this.state.beneficialOwners, (ownerKey) => {
+                // eslint-disable-next-line rulesdir/prefer-early-return
+                _.each(requiredFields, (inputKey) => {
+                    if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerKey}.${inputKey}`])) {
+                        const errorKey = errorKeys[inputKey] || inputKey;
+                        errors[`beneficialOwner.${ownerKey}.${inputKey}`] = this.props.translate(`bankAccount.error.${errorKey}`);
+                    }
+                });
+
+                if (values[`beneficialOwner.${ownerKey}.dob`] && !ValidationUtils.meetsAgeRequirements(values[`beneficialOwner.${ownerKey}.dob`])) {
+                    errors[`beneficialOwner.${ownerKey}.dob`] = this.props.translate('bankAccount.error.age');
+                }
+
+                if (values[`beneficialOwner.${ownerKey}.ssnLast4`] && !ValidationUtils.isValidSSNLastFour(values[`beneficialOwner.${ownerKey}.ssnLast4`])) {
+                    errors[`beneficialOwner.${ownerKey}.ssnLast4`] = this.props.translate('bankAccount.error.ssnLast4');
+                }
+
+                if (values[`beneficialOwner.${ownerKey}.street`] && !ValidationUtils.isValidAddress(values[`beneficialOwner.${ownerKey}.street`])) {
+                    errors[`beneficialOwner.${ownerKey}.street`] = this.props.translate('bankAccount.error.addressStreet');
+                }
+
+                if (values[`beneficialOwner.${ownerKey}.zipCode`] && !ValidationUtils.isValidZipCode(values[`beneficialOwner.${ownerKey}.zipCode`])) {
+                    errors[`beneficialOwner.${ownerKey}.zipCode`] = this.props.translate('bankAccount.error.zipCode');
+                }
+            });
         }
 
-        const errors = {};
-        _.each(this.requiredFields, (inputKey) => {
-            if (ValidationUtils.isRequiredFulfilled(this.state[inputKey])) {
-                return;
-            }
+        if (!ValidationUtils.isRequiredFulfilled(values.acceptTermsAndConditions)) {
+            errors.acceptTermsAndConditions = this.props.translate('common.error.acceptedTerms');
+        }
 
-            errors[inputKey] = true;
-        });
-        BankAccounts.setBankAccountFormValidationErrors({...errors, beneficialOwnersErrors});
-        return _.every(beneficialOwnersErrors, _.isEmpty) && _.isEmpty(errors);
+        if (!ValidationUtils.isRequiredFulfilled(values.certifyTrueInformation)) {
+            errors.certifyTrueInformation = this.props.translate('beneficialOwnersStep.error.certify');
+        }
+
+        return errors;
     }
 
-    removeBeneficialOwner(beneficialOwner) {
+    /**
+     * @param {int} ownerKey - ID connected to the beneficial owner identity form
+     */
+    removeBeneficialOwner(ownerKey) {
```

Comment:
**question:** Do we need to do anything in particular to stop the internal tracking of the keys in the `Form` component? Or does it not really matter?

Reply:
Oh I guess I would expect there to be some kind of clean up once the form is submitted, but maybe it's a question that applies to all forms.

Reply:
I think this is a nab for me so going to resolve but don't really love the idea of old stale values floating around that will never be used. Curious to get @luacmartins thoughts on whether this is something we need to worry about.

Reply:
Yeah, I agree it's not ideal. I guess on a successful submission we could clear all the past form values? Or when deleting a beneficial owner form remove any keys containing the ownerID?

Reply:
Yea, that was one of my concerns with this method. I think that we can remove these lingering keys when we remove beneficial owners.

Reply:
Or once the form is submitted there is a "clean up" step as presumably the old values would no longer be needed.

Reply:
@luacmartins I think I like this idea less because it adds an external requirement that anytime we have "dynamic" fields the author needs to remember that they must "cleanup" whenever fields are removed.

Reply:
Cleaning up on submit works too!